app/code/Magento/ImportExport/Model/ResourceModel/Import/Data/Iterator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\ImportExport\Model\ResourceModel\Import\Data;
+
+/**
+ * Import data iterator
+ */
+class Iterator extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
```

Comment:
Looks like this iterator may be decomposed into actual iterator and resource model which provides data to the iterator. Current implementation has multiple responsibilities which does not follow SOLID principles.